processor/src/main/java/org/mapstruct/ap/internal/processor/DefaultModelElementProcessorContext.java

```diff
@@ -68,7 +71,7 @@ public DefaultModelElementProcessorContext(ProcessingEnvironment processingEnvir
     }
 
     @Override
-    public Filer getFiler() {
+    public XFiler getFiler() {
```

Comment:
the XFiler API is quite different to the Filer API, but it should be able to do the same. One notable difference is that it seems to require a source (guessing for performance reasons), which might require a lot of parameters to be added where the source is far from the Filer. It might also be possible to start off without a source at the start and add this performance improvement later